service/update.go

```diff
@@ -96,3 +101,55 @@ func (service *UpdateServiceImp) GetConceptUpdate(ctx context.Context, conceptId
 		return conceptUpdate, http.StatusOK
 	}
 }
+
+// function to get a diff from database
+func (service *UpdateServiceImp) GetConceptUpdateDiff(ctx context.Context, conceptId string, updateId string) (*model.UpdateDiff, int) {
+	conceptUpdate, err := service.ConceptsRepository.GetConceptUpdate(ctx, conceptId, updateId)
+	if err != nil {
+		logrus.Error("Unable to get concept update")
+		return nil, http.StatusInternalServerError
+	}
+	if conceptUpdate == nil {
+		logrus.Error("Concept update not found")
+		return nil, http.StatusNotFound
+	}
+	query := bson.D{}
+	query = append(query, bson.E{Key: "resourceId", Value: conceptId})
+	query = append(query, bson.E{Key: "datetime", Value: bson.M{"$lt": conceptUpdate.DateTime}})
+	databaseUpdates, listErr := service.ConceptsRepository.GetConceptUpdates(ctx, query)
+	if listErr != nil {
+		logrus.Error("Get concept updates failed")
+		return nil, http.StatusInternalServerError
+	}
+	concept, err := service.BuildConceptFromPatches(databaseUpdates)
+	if err != nil {
+		logrus.Error("Unable to build concept from patches")
+		logrus.Error(err)
+		return nil, http.StatusInternalServerError
+
+	}
+	return &model.UpdateDiff{
+		ResourceId: conceptUpdate.ResourceId,
+		Operations: conceptUpdate.Operations,
+		Resource:   string(concept),
+	}, http.StatusOK
+}
+
+func (service *UpdateServiceImp) BuildConceptFromPatches(databaseUpdates []*model.UpdateDbo) ([]byte, error) {
+	concept := []byte("{}")
+	var err error
+
+	for _, update := range databaseUpdates {
+		if update == nil {
+			logrus.Warning("Update is nil")
+		} else {
+			logrus.Info("Update: " + update.ID)
+			concept, err = applyPatchesToResource(concept, *update)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return concept, nil
```

Comment:
Tror denne kan flyttes inn i utils:
```suggestion
func BuildResourceFromPatches(databaseUpdates []*model.UpdateDbo) ([]byte, error) {
	resource := []byte("{}")
	var err error

	for _, update := range databaseUpdates {
		if update == nil {
			logrus.Warning("Update is nil")
		} else {
			logrus.Info("Update: " + update.ID)
			resource, err = applyPatchesToResource(resource, *update)
			if err != nil {
				return nil, err
			}
		}
	}
	return resource, nil
```